src/main/scala/nl/knaw/dans/easy/multideposit/actions/AddFileMetadataToDeposit.scala

```diff
@@ -17,32 +17,103 @@ package nl.knaw.dans.easy.multideposit.actions
+    def resolve(filepath: String): File = new File(settings.multidepositDir, filepath).getAbsoluteFile
+
+    dataset.getColumns("AV_FILE", "AV_FILE_TITLE", "AV_SUBTITLES", "AV_SUBTITLES_LANGUAGE")
+      .toRows
+      .map(row => (
+        row.get("AV_FILE"),
+        row.get("AV_FILE_TITLE"),
+        row.get("AV_SUBTITLES"),
+        row.get("AV_SUBTITLES_LANGUAGE")
+      ))
+      .collect {
+        case (maybeFile, maybeTitle, maybeSub, maybeSubLang) if maybeFile.exists(!_.isBlank) =>
+          (resolve(maybeFile.get), maybeTitle, maybeSub.map(resolve), maybeSubLang)
+      }
+      .groupBy { case (file, _, _, _) => file }
+      .map { case (file, instr) =>
+        val optTitle = instr.collectFirst { case (_, Some(title), _, _) => title }
+        val subtitles = instr.collect { case (_, _, Some(sub), maybeSubLang) => Subtitle(sub, maybeSubLang) }
+        file -> FileInstruction(file, optTitle, subtitles)
+      }
+  }
+
+  private lazy val avAccessibility: FileAccessRights.Value = {
+    dataset.findValue("SF_ACCESSIBILITY")
+      .flatMap(FileAccessRights.valueOf)
+      .orElse(dataset.findValue("DDM_ACCESSRIGHTS")
+        .map(v => FileAccessRights.accessibleTo(AccessCategory.valueOf(v))))
+      .getOrElse(throw ActionException(row, "No value found for either SF_ACCESSIBILITY or DDM_ACCESSRIGHTS"))
+  }
+
+  private def getFileMetadata(file: File): Try[FileMetadata] = {
+    def mkFileMetadata(m: MimeType, voca: AudioVideo): AVFileMetadata = {
+      fileInstructions.get(file) match {
+        case Some(FileInstruction(_, optTitle, subtitles)) =>
+          val title = optTitle.getOrElse(file.getName)
+          AVFileMetadata(file, m, voca, title, avAccessibility, subtitles)
+        case None =>
+          AVFileMetadata(file, m, voca, file.getName, avAccessibility, Seq.empty)
+      }
+    }
+
+    getMimeType(file).map {
+      case mimeType if mimeType startsWith "audio" => mkFileMetadata(mimeType, Audio)
+      case mimeType if mimeType startsWith "video" => mkFileMetadata(mimeType, Video)
+      case mimeType => new FileMetadata(file, mimeType)
+    }
+  }
+
+  private lazy val fileMetadata: Try[Seq[FileMetadata]] = Try {
     val datasetDir = multiDepositDir(datasetID)
     if (datasetDir.exists()) {
-      val files = datasetDir.listRecursively
-
-      files.map(getMimeType)
+      datasetDir.listRecursively
+        .map(file => getFileMetadata(file.getAbsoluteFile))
         .collectResults
-        .map(files.zip(_))
+        .recoverWith {
+          case CompositeException(es) =>
+            // filter duplicate messages
+            Failure(CompositeException(es.foldRight(List.empty[Throwable])((t, ts) => {
+              val msg = t.getMessage
+              ts.find(_.getMessage == msg).fold(t :: ts)(_ => ts)
+            })))
+          case e => Failure(e)
+        }
     }
     else // this means the dataset does not contain any data
-      Try { List.empty }
+      Success { List.empty }
```

Comment:
you mean `/<output-directory>/bag/data`? That is take care of by another action. The '_dataset_' mentioned here is the input.